### PR TITLE
fix: Validate variant in query against question

### LINKED ISSUE #5360: Instance question rendering with incorrect question
Cinda reported an issue on Slack where students were seeing question render errors for which the stack trace referenced elements that weren't even used in the question. I'm going to capture what I find during investigation here.

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Fixes #5360

### CODE REVIEW

File: sprocs/variants_select.sql [hunk 1]
```diff
@@ -20,9 +22,13 @@ BEGIN
         LEFT JOIN instance_questions AS iq ON (iq.id = v.instance_question_id)
         LEFT JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
         LEFT JOIN assessments AS a ON (a.id = ai.assessment_id)
-    WHERE v.id = variants_select.variant_id;
+    WHERE v.id = variants_select.variant_id
+          AND v.question_id = variants_select.question_id
+          -- instance_question_id is null for question preview, so allow any variant of the question
+          AND (variants_select.instance_question_id IS NULL
+               OR v.instance_question_id = variants_select.instance_question_id);
```
Comment (nwalters512) [inline sprocs/variants_select.sql:29]: Formatting nit:
```suggestion
    WHERE
        v.id = variants_select.variant_id
        AND v.question_id = variants_select.question_id
        -- instance_question_id is null for question preview, so allow any variant of the question
        AND (
            variants_select.instance_question_id IS NULL
            OR v.instance_question_id = variants_select.instance_question_id
        );
```

File: sprocs/variants_select.sql [hunk 2]
```diff
@@ -20,9 +22,13 @@ BEGIN
         LEFT JOIN instance_questions AS iq ON (iq.id = v.instance_question_id)
         LEFT JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
         LEFT JOIN assessments AS a ON (a.id = ai.assessment_id)
-    WHERE v.id = variants_select.variant_id;
+    WHERE v.id = variants_select.variant_id
+          AND v.question_id = variants_select.question_id
```
Comment (nwalters512) [inline sprocs/variants_select.sql:26]: Per the docs for `variants_insert`, `question_id` can be null. I'm not sure if that's actually true anymore (I suspect it actually isn't), but that's probably worth spending a small amount of time looking into.

**Review by nwalters512 — APPROVED** [review]

File: sprocs/variants_select.sql (hunk 2 above)
Comment (jonatanschroeder) [reply sprocs/variants_select.sql]: Resolved in 9d7f050.

**Comment by cheeren** [comment]
Was this deployed on the Canadian server? I received another report from cpsc317 -- it looks like the latest occurrences were 13hr ago (i don't have student data access). Apparently one of the people seeing these strange messages is a TA, so they're a little worried someone is snooping around.

**Comment by jonatanschroeder** [comment]
> Was this deployed on the Canadian server? I received another report from cpsc317 -- it looks like the latest occurrences were 13hr ago (i don't have student data access). Apparently one of the people seeing these strange messages is a TA, so they're a little worried someone is snooping around.

@mwest1066 deployed it this morning.

**Pull request merged**